src/web/services/search.service.ts

```diff
@@ -167,17 +155,17 @@ export class SearchService {
   }
 
   getPrivileges(
-    coursesWithSections: SearchStudentsTable[],
+    coursesWithStudents: SearchStudentsListRowTable[],
   ): Observable<InstructorPrivilege[]> {
-    if (coursesWithSections.length === 0) {
+    if (coursesWithStudents.length === 0) {
       return of([]);
     }
     return forkJoin(
-      coursesWithSections.map((course: SearchStudentsTable) => {
-        return course.sections.map((section: StudentListSectionData) => {
+      coursesWithStudents.map((course: SearchStudentsListRowTable) => {
+        return course.students.map((studentModel: StudentListRowModel) => {
           return this.instructorService.loadInstructorPrivilege({
```

Comment:
It is too expensive to load section for every student. Either you want to group them first to load privilege for one section or use some cache variable to keep the privilege already loaded. Maybe @t-cheepeng can help.

Reply:
I think it would be easier to group the students in the same section together then load the privileges.
@edwardkang0925 Please take note and try to make the changes for this! If you need any help please don't hesitate to ask

Reply:
@t-cheepeng I will work on it and ask question if I get stuck. Thank you!

Reply:
@t-cheepeng  I tried to group the students with the same section name, but I couldn't come up with an efficient grouping method, so I used double for-loop to process the data. While doing so, I used a cache variable to keep distinct privileges of each course and assign the cached privileges to each student enrolled in the same course with same section. By doing so, I think I prevented loading duplicated privileges from the back-end. However, I am not confident whether this approach slows down the performance without my acknowledgment. Could you review my modification? Thank you